### PR TITLE
STM32WBA  Cube 1.4.1 release alignment

### PR DESCRIPTION
**Pull request by asm5878**
STM32WBA Cube 1.4.1 release to be integrated

### CODE REVIEW

File: stm32cube/stm32wbaxx/README [hunk 1]
```diff
@@ -35,6 +35,7 @@ License Link:
    https://opensource.org/licenses/BSD-3-Clause
 
 Patch List:
+--> please check that the following list is still valid:
```
Comment (erwango) [inline stm32cube/stm32wbaxx/README]: This line could be removed.

File: lib/stm32wba/hci/ll_intf_cmn.h
Comment (erwango) [inline lib/stm32wba/hci/ll_intf_cmn.h:1]: Quick question on this file. Did you apply dos2unix and removed trailing white spaces ?

File: stm32cube/stm32wbaxx/README [hunk 2]
```diff
@@ -49,4 +48,7 @@ Patch List:
     Impacted files:
      drivers/include/Legacy/stm32_hal_legacy.h
 
+   *Including common interfaces of MAC only and BLE/COEXISTENCE APIs
+    -Added lib/stm32wba/hci/ll_intf_cmn.h
+
```
Comment (erwango) [inline stm32cube/stm32wbaxx/README:53]: This is the HAL README, please update the lib/stm32wba/hci/README instead, there is already a list of the files included.

**Review by erwango — CHANGES_REQUESTED** [review]
Minor comments, otherwise LGTM

**Comment by asm5878** [comment]
PR updated according your inputs, could you please review ?

File: stm32cube/stm32wbaxx/README (hunk 1 above)
Comment (asm5878) [reply stm32cube/stm32wbaxx/README]: ok, done

File: lib/stm32wba/hci/ll_intf_cmn.h
Comment (asm5878) [reply lib/stm32wba/hci/ll_intf_cmn.h:1]: Applied and updated

File: stm32cube/stm32wbaxx/README (hunk 2 above)
Comment (asm5878) [reply stm32cube/stm32wbaxx/README]: Done

File: lib/stm32wba/hci/README [hunk 3]
```diff
@@ -117,6 +117,9 @@ Patch List:
 			  app_conf.h
 			  scm.c
 
+	* Including common interfaces of MAC only and BLE/COEXISTENCE APIs
+	  Added file: lib/stm32wba/hci/ll_intf_cmn.h
+
```
Comment (erwango) [inline lib/stm32wba/hci/README:122]: Instead of adding this, please update the file list above.

**Review by erwango — CHANGES_REQUESTED** [review]
One minor comment, otherwise LGTM

File: lib/stm32wba/hci/ll_sys_intf.c [hunk 4]
```diff
@@ -21,6 +21,9 @@
 #include "event_manager.h"
 #include "ll_intf.h"
 
+extern uint8_t AHB5_SwitchedOff;
+extern uint32_t radio_sleep_timer_val;
```
Comment (erwango) [inline lib/stm32wba/hci/ll_sys_intf.c:25]: This can't work. You're not supposed to use Zephyr code here `uint32_t/uint8_t`.
This should be Cube code only.

**Review by erwango — CHANGES_REQUESTED** [review]

File: lib/stm32wba/hci/README (hunk 3 above)
Comment (asm5878) [reply lib/stm32wba/hci/README]: done

File: lib/stm32wba/hci/linklayer_plat.c [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+
+#include "scm.h"
+
+#define LOG_LEVEL CONFIG_SOC_LOG_LEVEL
+LOG_MODULE_REGISTER(linklayer_plat);
```
Comment (erwango) [inline lib/stm32wba/hci/linklayer_plat.c:12]: You're not supposed to use Zephyr code in this module.

**Review by erwango — CHANGES_REQUESTED** [review]

File: lib/stm32wba/hci/README [hunk 6]
```diff
@@ -116,6 +116,9 @@ Patch List:
 			  main.h
 			  app_conf.h
 			  scm.c
+			  ll_intf_cmn.h
+			  ll_sys_if.c
+			  linklayer_plat.c
```
Comment (erwango) [inline lib/stm32wba/hci/README:121]: Please update file list above (line 17) to explain where these files are coming from and update script

File: lib/stm32wba/hci/README [hunk 7]
```diff
@@ -116,7 +118,11 @@ Patch List:
...
 
 	* Changes from official delivery:
 	- dos2unix applied
 	- trailing white spaces removed
+	- #ifndef __ZEPHYR__ applied to remove Cube specific implementation
```
Comment (erwango) [inline lib/stm32wba/hci/README:128]: I'd prefer a dedicated bullet for this point which helps identifying the impacted files:
```
	* #ifndef __ZEPHYR__ applied to remove Cube specific implementation
	  Impacted file:  ll_sys_if.c
	                         linklayer_plat.c
```

File: lib/stm32wba/hci/README [hunk 8]
```diff
@@ -74,6 +74,8 @@ Description:
       Projects/NUCLEO-WBA55CG/Applications/BLE/BLE_HeartRate/STM32_WPAN/Target/power_table.c
       Projects/NUCLEO-WBA55CG/Applications/BLE/BLE_HeartRate/STM32_WPAN/Target/bpka.c
       Projects/NUCLEO-WBA55CG/Applications/BLE/BLE_HeartRate/STM32_WPAN/Target/bpka.h
+      Projects/NUCLEO-WBA55CG/Applications/BLE/BLE_HeartRate/STM32_WPAN/Target/linklayer_plat.c
+      Projects/NUCLEO-WBA55CG/Applications/BLE/BLE_HeartRate/STM32_WPAN/Target/ll_sys_if.c
```
Comment (erwango) [inline lib/stm32wba/hci/README:78]: ll_intf_cmn.h is missing

**Review by erwango — CHANGES_REQUESTED** [review]
Some minor changes required otherwise LGTM

File: lib/stm32wba/hci/README (hunk 7 above)
Comment (asm5878) [reply lib/stm32wba/hci/README]: Fixed

File: lib/stm32wba/hci/README (hunk 8 above)
Comment (asm5878) [reply lib/stm32wba/hci/README:79]: Fixed

**Comment by asm5878** [comment]
Just additional clean up

File: lib/stm32wba/hci/README [hunk 9]
```diff
@@ -116,6 +119,13 @@ Patch List:
 			  main.h
 			  app_conf.h
 			  scm.c
+			  ll_intf_cmn.h
+			  ll_sys_if.c
+			  linklayer_plat.c
+
+	* #ifndef __ZEPHYR__ applied to remove Cube specific implementation
+	  Impacted files: ll_sys_if.c
+					  linklayer_plat.c
```
Comment (erwango) [inline lib/stm32wba/hci/README:128]: Please fix indentation

File: lib/stm32wba/hci/linklayer_plat.c
Comment (erwango) [inline lib/stm32wba/hci/linklayer_plat.c:1]: This whole file differs from Cube file in some areas, like indentation. Or `LINKLAYER_PLAT_EnableIRQ/LINKLAYER_PLAT_DisableIRQ` functions.
So at next version update, there will be changes unrelated to file evolution.

Ideally, this should be exctaly the same with only some `#ifndef __ZEPHYR__`

Comment (asm5878) [reply lib/stm32wba/hci/linklayer_plat.c:1]: Thanks for the remark, now it should be fine.

**Review by erwango — APPROVED** [review]
